packages/agent/src/app.ts

```diff
@@ -209,10 +244,28 @@ export class App {
     const endpoints = await Promise.all(endpointPromises);
     this.validateAgentEndpoints(channels, endpoints);
 
-    // Remove all channels from list that are present in the new definition
-    // We will remove the channels that are left over -- channels that are not part of the new config
-    for (const definition of channels) {
-      pendingRemoval.delete(definition.name);
+    const filteredChannels = [] as AgentChannel[];
```

Comment:
Doesn't look like `filteredChannels` or `filteredEndpoints` is used anywhere?

Reply:
Ah you're right. I think there's technically a bug here, and we should be using the filtered arrays instead of only filtering channels. Going to add a test for multiple channels to catch this